helix-term/src/ui/editor.rs

```diff
@@ -374,12 +406,24 @@ impl EditorView {
                             }
                         } else {
                             let grapheme = Cow::from(grapheme);
+                            let is_whitespace;
 
                             let (grapheme, width) = if grapheme == "\t" {
+                                is_whitespace = true;
                                 // make sure we display tab as appropriate amount of spaces
                                 let visual_tab_width = tab_width - (visual_x as usize % tab_width);
-                                (&tab[..visual_tab_width], visual_tab_width)
+                                let grapheme_tab_width = tab
```

Comment:
Visual tab is a char so it's guaranteed to be a single grapheme, no? You can use the char_indices() iter from std. Even faster would be using https://docs.rs/ropey/latest/ropey/str_utils/fn.char_to_byte_idx.html